subprojects/core/src/main/java/org/gradle/api/tasks/AbstractExecTask.java

```diff
@@ -360,7 +371,21 @@ void setExecAction(ExecAction execAction) {
      * @return The result. Returns {@code null} if this task has not been executed yet.
      */
     @Internal
+    @Deprecated
```

Comment:
in general, we shouldn't deprecate something whose replacement is incubating.

Reply:
Should there be a difference between "soft" deprecation (API/source code level) and nagging deprecations?  I can remove the nagging for now, but the `@Deprecated` might be useful to move people to the new API.